src/mumble/AudioInput.h

```diff
@@ -138,6 +138,7 @@ class AudioInputRegistrar {
 	AudioInputRegistrar(const QString &n, int priority = 0);
 	virtual ~AudioInputRegistrar();
 	virtual AudioInput *create()                               = 0;
+	virtual const QVariant getDeviceChoice()                   = 0;
```

Comment:
Why is this using a `QVariant`?

Reply:
It's the data type that is used as key in `audioDevice`.

The value is a `QString`.

Reply:
Using a variant for something that always is a String seems out-of-place though :eyes:

Reply:
It's not always a string. The PortAudio backend uses an integer, for example:

https://github.com/mumble-voip/mumble/blob/c54363aa982d65c11b6aef4cc3264c2b08ec4f78/src/mumble/Settings.h#L230

Reply:
Ah okay then :+1: